### DISCUSSION TITLE
csv files

### DISCUSSION THREAD

**Discussion by NDetenbeck** [git-lfs/git-lfs#5953] [Q&A]
git-lfs is adding .txt files but not .csv file

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.

Would you be able to provide some more information about your situation?  The commands you've used and any error messages you've seen, and the operating system you're using are all important details.  For instance, do your OS and filesystem treat filenames case-sensitively or case-insensitively?  (If the former, you may need to use a file extension pattern which matches both lowercase and uppercase characters.)

Also, do you have CSV files in your Git history already, or not?  The answer we give depends on whether that is the case or not.

#### No Pre-Existing Files

If you don't have any CSV files in your Git repository's history, you can just run `git lfs track "*.csv"` and commit the change to the `.gitattributes` file.  Any files you add with the `.csv` extension after that will be treated as Git LFS objects.

#### Pre-Existing Files

If you do have existing files with the `.csv` extension in your Git commit history, you can either convert them to Git LFS objects starting with a new commit, or rewrite your Git history so they are always treated as Git LFS objects.  I'll discuss these two approaches separately.

- If you're comfortable with a point-in-time commit before which the CSV files are regular Git objects, and after which the files are Git LFS objects, you should be able to run `git lfs track "*.csv"`.  Then all the existing files with that extension should be marked as modified, so a subsequent `git add` on them will convert them to Git LFS objects.  Assuming you don't have any other changes in your working tree (i.e., other files you don't want to add to Git right now), something like the following should work:

    ```
    $ git lfs track "*.csv"
    $ git status
    # This should show the change to .gitattributes and also show all *.csv files as modified.

    $ git add .
    # This should add both .gitattributes and all the *.csv files to the Git index.

    $ git commit -m "Convert CSV files to Git LFS"
    ```

- If you prefer to convert all the files matching the `*.csv` pattern throughout your Git history, you can use instead a command like:
    ```
    $ git lfs migrate import --include="*.csv" --everything
    ```

    That will rewrite your full Git history, on all branches, so that all files matching the `.csv` extension will be Git LFS objects.  (The `--everything` option means to rewrite all branches.)  However, note that you will now have new, different SHA values for every Git commit after the first Git LFS object is added on any branch!  This means you cannot expect to reference commits using the SHAs you've used in the past, and you will likely need to force-push your changes to any Git remotes, and any other people who are using your repository should re-clone it afterwards.  So this step can be a major and disruptive change; if you are working on a project with other people, take care and make sure you understand all the steps and consequences before proceeding with this approach.
